packages/hydrooj/src/handler/user.ts

```diff
@@ -471,6 +471,11 @@ class OauthCallbackHandler extends Handler {
             this.session.scope = PERM.PERM_ALL.toString();
             this.response.redirect = '/';
         } else {
+            if (this.user.hasPriv(PRIV.PRIV_USER_PROFILE)) {
```

Comment:
This logic might lead to unwanted behavior.
Binding an oauth account (with different email and unknown openId) is a dangerous operation, this should be a sudo-required operation and user should be notified (via email, etc).
And be sure to add a page allowing user to manage (or delete) any previously binded accounts.

Reply:
I have discussed the logic related to OAuth with panda, but when deleting OpenID, it was not possible to distinguish between certain types of OAuth (e.g. Github and Google). Except for single user one click unbinding, a single unbinding operation cannot be completed on the existing database